src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -11,8 +11,10 @@ const TodoHeader: FC = () => {
 
   const submitValue = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
-    dispatch(addTodo(input));
-    setInput("");
+    if (input.trim() !== "") {
+      dispatch(addTodo(input));
```

Comment:
as we know the empty string is also falsy, you can write "!input.trim()"

Reply:
If use !input.trim() instead of input.trim() !== "", it turns out that you can only add empty lines with a space by pressing enter, or I didn’t understand

Reply:
I pick it, you mean add input.trim() only so it work exactly the same

Reply:
sorry, yeap

Reply:
just if (input.trim())